### PR TITLE
add lab2 kra

### PR DESCRIPTION
**Pull request by anstkras**

### CODE REVIEW

**Comment by anstkras** [comment]
https://stepik.org/submissions/218135/105894333?unit=32920

File: solutions/kra/lab2/kern/pmap.c [hunk 1]
```diff
@@ -98,8 +98,17 @@ boot_alloc(uint32_t n)
 	// to a multiple of PGSIZE.
 	//
 	// LAB 2: Your code here.
+	if (n == 0) {
+		return nextfree;
+	}
+
+	if((uint32_t)nextfree - KERNBASE >= 0xfffffff) {
```
Comment (krinkin) [inline solutions/kra/lab2/kern/pmap.c]: Please explain this magic number

**Review by krinkin — CHANGES_REQUESTED** [review]

File: solutions/kra/lab2/kern/pmap.c (hunk 1 above)
Comment (anstkras) [reply solutions/kra/lab2/kern/pmap.c]: 4 Gb

Comment (krinkin) [reply solutions/kra/lab2/kern/pmap.c]: 4Gb=0xffffffff

Comment (anstkras) [reply solutions/kra/lab2/kern/pmap.c]: Точно, исправлено

**Review by krinkin — APPROVED** [review]

**Pull request merged**